src/pgzero/storage.py

```
@@ -23,17 +37,33 @@ def _get_platform_pgzero_path():
 
     """
     if platform.system() == 'Windows':
-        try:
-            appdata = os.environ['APPDATA']
-        except KeyError:
-            raise KeyError(
-                "Couldn't find the AppData directory for Pygame Zero save "
-                "data. Please set the %APPDATA% environment variable."
-            )
+        appdata = _get_windows_environ("APPDATA")
         return os.path.join(appdata, 'pgzero')
     return os.path.expanduser(os.path.join('~', '.config/pgzero/saves'))
 
 
+def _get_platform_screenshot_path():
+    r"""Get the screenshot directory for pgzero.
+
+    Under Windows, this is %USERPROFILE%\Pictures\pgzero. Under Linux/MacOS,
+    it's ~/Pictures/pgzero."""
+    if platform.system() == "Windows":
+        home = _get_windows_environ("USERPROFILE")
+        return os.path.join(home, "Pictures", "pgzero")
+    return os.path.expanduser(os.path.join("~", "Pictures", "pgzero"))
+
+
+# Moved this function out of Storage to make it reusable.
+def _ensure_path(path):
+    """Checks if the path exists and makes all necessary directories if not."""
+    try:
+        os.makedirs(path)
+    except IsADirectoryError:
+        pass
+    except FileExistsError:
+        pass
```

I think it's odd for these errors to pass silently. It might make sense for storage(?) but it makes less sense for screenshots. We should just let the error propagate.

---

Fixed. The errors were passed because `os.makedirs()` throws an error by default if the directory already exists. Since this is just there to make sure it does and create folders if not, the errors should not create a crash.

I've left `storage.py` the same, but the screenshot functionality now uses the built-in parameter `exist_ok` on `os.makedirs()` to make this clearer.